src/app/fdctl/run/tiles/fd_replay.c

```diff
@@ -520,19 +588,74 @@ after_frag( void *             _ctx,
+    /*                                                 ctx->gossip_out_chunk0, ctx->gossip_out_wmark ); */
+
+    /* TEMPORARY: echo vote txns from gossip to TPU; replace by our own vote txns later */
+    ulong               slot          = vote.root + 50; /* This is a strawman implementation and we will use PoH for current slot  */
+    /* Get the leader contact info of slot */
+    fd_stake_ci_t * stake_ci          = ctx->stake_ci;
+    fd_epoch_leaders_t const * lsched = fd_stake_ci_get_lsched_for_slot( stake_ci, slot );
+    fd_shred_dest_t *           sdest = fd_stake_ci_get_sdest_for_slot ( stake_ci, slot );
+    if ( !lsched || !sdest ) return;
+    fd_pubkey_t const *   slot_leader = fd_epoch_leaders_get( lsched, slot );
+    fd_shred_dest_idx_t      dest_idx = fd_shred_dest_pubkey_to_idx( sdest, slot_leader );
+    fd_shred_dest_weighted_t *   dest = fd_shred_dest_idx_to_dest( sdest, dest_idx );
+
+    if (dest->ip4 != 0 && dest->port != 0) {
+      /* Define packet buffer and generate vote transaction */
+      uchar txn_meta_buf [ FD_TXN_MAX_SZ ] ;
+      uchar * packet    = fd_chunk_to_laddr( ctx->net_out_mem, ctx->net_out_chunk );
+      ulong vote_txn_sz = fd_vote_txn_generate( &voter,
+                                                &vote,
+                                                ctx->gossip_vote_txn + recent_blockhash_off,
+                                                txn_meta_buf,
+                                                packet + sizeof(fd_net_hdrs_t) );
+
+      /* Generate packet headers */
+      fd_memcpy( packet, ctx->hdr, sizeof(fd_net_hdrs_t) );
+      fd_net_hdrs_t * hdr = (fd_net_hdrs_t *)packet;
+
+      memset( hdr->eth->dst, 0U, 6UL );
+      memcpy( hdr->ip4->daddr_c, &dest->ip4, 4UL );
+      hdr->ip4->net_id      = fd_ushort_bswap( ctx->net_id++ );
+      hdr->ip4->check       = 0U;
+      hdr->ip4->net_tot_len = fd_ushort_bswap( (ushort)(vote_txn_sz + sizeof(fd_ip4_hdr_t)+sizeof(fd_udp_hdr_t)) );
+      hdr->ip4->check       = fd_ip4_hdr_check( ( fd_ip4_hdr_t const *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4 ) );
+
+      ulong packet_sz       = sizeof(fd_net_hdrs_t) + vote_txn_sz;
+      hdr->udp->net_dport   = fd_ushort_bswap( dest->port );
+      hdr->udp->net_len     = fd_ushort_bswap( (ushort)(vote_txn_sz + sizeof(fd_udp_hdr_t)) );
+      hdr->udp->check       = fd_ip4_udp_check( *(uint *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4->saddr_c ),
+                                                *(uint *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4->daddr_c ),
+                                                (fd_udp_hdr_t const *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->udp ),
+                                                packet + sizeof(fd_net_hdrs_t) );
+
+      /* Send the vote transaction through net tile */
+      ulong tspub = fd_frag_meta_ts_comp( fd_tickcount() );
+      ulong tsorig = fd_frag_meta_ts_comp( fd_tickcount() );
```

Comment:
can you put all this in a function `send_vote` in the replay tile that takes in the tile ctx and vote_txn as arguments ?

Reply:
Done. The code is much cleaner now.